src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -355,92 +355,170 @@ class Config:
+        if not data.get('user'):
+            verrors.add(f'{schema}.user', 'This field is required')
+            raise verrors
+
+        username = data.get('user')
+        if ' ' in username:
+            verrors.add(f'{schema}.user', 'User names cannot have spaces')
+            raise verrors
+
+        user = await self.middleware.call(
+            'notifier.get_user_object',
+            username
+        )
+        if not user:
+            verrors.add(f'{schema}.user', f'Provided user "{username}" does not exist')
+            raise verrors
+
+        remote_host = data.get('remotehost')
+        if not remote_host:
+            verrors.add(f'{schema}.remotehost', 'Please specify a remote host')
 
-        if data.get("extra"):
-            data["extra"] = " ".join(data["extra"])
+        if data.get('extra'):
+            data['extra'] = ' '.join(data['extra'])
         else:
-            data["extra"] = ""
+            data['extra'] = ''
 
-        if data.get("month"):
+        if data.get('month'):
             if len(data['month']) == 12:
                 data['month'] = '*'
             else:
-                data['month'] = ",".join(data['month'])
+                data['month'] = ','.join(data['month'])
 
-        if data.get("dayweek"):
+        if data.get('dayweek'):
             if len(data['dayweek']) == 7:
                 data['dayweek'] = '*'
             else:
                 data['dayweek'] = ",".join(data['dayweek'])
 
-        mode = data.get("mode")
-        rmodule = data.get("remotemodule")
-        if mode == 'module' and not rmodule:
-            verrors.add('remotemodule', 'This field is required')
+        mode = data.get('mode')
+        if not mode:
+            verrors.add(f'{schema}.mode', 'This field is required')
 
-        rpath = data.get("remotepath")
-        if mode == 'ssh' and not rpath:
-            verrors.add('remotepath', 'This field is required')
+        remote_module = data.get('remotemodule')
+        if mode == 'module' and not remote_module:
+            verrors.add(f'{schema}.remotemodule', 'This field is required')
 
-        username = data.get("user")
         if mode == 'ssh':
-            rport = data.get("remoteport")
-            if not rport:
-                verrors.add('remoteport', 'This field is required')
-  
-            try:
-                user = await self.middleware.call('user.query', [('username', '=', username)], {'get': True})
-            except IndexError:
-                verrors.add('user', 'Provided user does not exists')
+            remote_port = data.get('remoteport')
+            if not remote_port:
+                verrors.add(f'{schema}.remoteport', 'This field is required')
+
+            remote_path = data.get('remotepath')
+            if not remote_path:
+                verrors.add(f'{schema}.remotepath', 'This field is required')
+
+            search = os.path.join(user.pw_dir, '.ssh', 'id_[edr]*')
```

Comment:
This won't check the presence of private key, it will be true if only public key is present

Reply:
Kindly have a look now and let me know if it's good : )

Reply:
I agree, this is more neat this way. Thank you for pointing it out : ) Kindly let me know if it's okay now.